ops/model.py

```diff
@@ -1866,10 +1877,10 @@ class StatusBase:
     directly use the child class such as :class:`ActiveStatus` to indicate their status.
     """
 
-    _statuses: Dict[str, Type['StatusBase']] = {}
+    _statuses: Dict[_StatusName, Type['StatusBase']] = {}
 
-    # Subclasses should override this attribute
-    name = ''
+    # Subclasses must provide this attribute
+    name: _StatusName
```

Comment:
I think it's ok to make it public, and I think it would be good to do so. This is very much a read only property, so the narrower type is fine. No one should be setting this attribute, and no one should be subclassing this base class, especially not with a non-`StatusName` "name".

The following type check won't break since name is a union of literal strings.
```python
my_var: str = status_object.name
```

The signature of `StatusBase.from_name` hasn't changed -- the `name` argument is still typed as `str`. Ideally someday it would be typed as `StatusName`, but for now this isn't changed, so nothing should break.

I do think that I should update the `StatusBase.from_name` docstring to list all six valid names.

Reply:
Per discussion in daily sync, we'll make `StatusName` public.